synapse/util/caches/expiringcache.py

```diff
@@ -100,6 +98,18 @@ def __getitem__(self, key):
 
         return entry.value
 
+    def pop(self, key, default=None):
+        value = self._cache.pop(key, SENTINEL)
+        if value is SENTINEL:
+            return default
+
+        if self.iterable:
```

Comment:
I think the problem is that we're potentially conflating "eviction because of invalidation" (which this code *may* do) with "eviction because of some automated process" in the statistics. If something pops from the cache so it won't be cached anymore deliberately, I don't think that's useful to track in what this metric appears to do (assessing whether our cache is effective).

Reply:
@hawkowl: so I think you're agreeing with me? If so I'll remove this bit of code and merge it.

Reply:
Ah, for some reason I thought that this was consistent with what we did elsewhere, but it isn't.